mgail.py

```diff
@@ -75,28 +92,52 @@ def __init__(self, environment):
 
         # 2. Discriminator
         labels = tf.concat([1 - self.label, self.label], 1)
-        d = self.discriminator.forward(states, actions)
-
-        # 2.1 0-1 accuracy
-        correct_predictions = tf.equal(tf.argmax(d, 1), tf.argmax(labels, 1))
-        self.discriminator.acc = tf.reduce_mean(tf.cast(correct_predictions, "float"))
-        # 2.2 prediction
-        d_cross_entropy = tf.nn.softmax_cross_entropy_with_logits(logits=d, labels=labels)
-        # cost sensitive weighting (weight true=expert, predict=agent mistakes)
-        d_loss_weighted = self.env.cost_sensitive_weight * tf.multiply(tf.compat.v1.to_float(tf.equal(tf.squeeze(self.label), 1.)), d_cross_entropy) +\
-                                                           tf.multiply(tf.compat.v1.to_float(tf.equal(tf.squeeze(self.label), 0.)), d_cross_entropy)
-        discriminator_loss = tf.reduce_mean(d_loss_weighted)
-        self.discriminator.train(objective=discriminator_loss)
+        lprobs = self.lprobs
+        
+        # MODIFIED DISCRIMINATOR SECTION
+        if self.use_irl:
+            self.discrim_output, log_p_tau, log_q_tau, log_pq = self.discriminator.forward(states_, actions, states, lprobs)
+
+
+            correct_predictions = tf.equal(self.discrim_output, tf.argmax(labels, 1))
```

Comment:
```suggestion
            correct_predictions = tf.equal(tf.round(self.discrim_output), tf.argmax(labels, 1))
```
Should we round the discriminator output before comparing them? coz the output should be a scalar ~ [0, 1]? Not sure how to measure it for continuous action though

Reply:
Good catch! Since our discriminator in the IRL case is using a single dim output (1) instead of 2, we'll need to do a round.

Reply:
Addressed in https://github.com/yipsang/mgail/pull/7/commits/e9397141abebae3119b02e5f33ebfcac924c154e